components/select/Select.razor.cs

```diff
@@ -488,6 +497,14 @@ public override TItemValue Value
                 if (_valueHasChanged)
                 {
                     _selectedValue = value;
+                    if (_isInitialized)
+                    {
+                        OnValueChange(value);
+                        if (Form?.ValidateOnChange == true)
+                        {
+                            EditContext?.NotifyFieldChanged(FieldIdentifier);
+                        }
+                    }
```

Comment:
I don't understand why you write code in the property setters. As I remember, the setters are called depend on the property handwriting order. If the order of code execution matters, then you should not write code in the property setters. Here we want the code in the `Value` setter to be called after the code in the `DataSource` setter, right?